week6/js/main.js

```diff
@@ -0,0 +1,97 @@
+const buttonEl = document.querySelector(".search-btn");
+
+buttonEl.addEventListener("click", (event) => {
+  const inputBoxEl = document.querySelector("#input-bar");
+  let keyword = inputBoxEl.value.toUpperCase();
+
+  if(keyword === '')
+    return alert('검색어를 입력하세요');
+  //console.log(keyword)
+  searching(keyword);
+});
+
+function searching(keyword) {
+
+  let date = new Date(); //코드 추가
+  let today = date.toLocaleDateString().replace(/\s/g,'').replace(/\./g,'-');
+  //console.log(today);
+
+  var url = 'https://newsapi.org/v2/everything?' +
+          `q=${keyword}&` +
+          `from=${today}&` +
+          'sortBy=popularity&' +
+          'apiKey=04324f28976e49bd84feaeb33b93ff3d';
+
+  var req = new Request(url);  
+
+  fetch(req)
+    .then(function(res) {
+        return res.json()
+    })
+    .then( (res) => {
+      console.log(res);
+
+      showArticle(res);
+    })
+}
+
+function showArticle(res) {
+  const aEl = document.querySelector('.sectionAnchor');
+  const imgEl = document.querySelector('img');
+  const authorEl = document.querySelector('small');
+  const dateEl = document.querySelector('section > span');
+  //console.log(dateEl)
+  //console.log(typeof res.articles[0].publishedAt)
+  const titleEl = document.querySelector('h2');
+  const contentEl = document.querySelector('section > p');
+
+  const articleLists = document.querySelector('.article-lists')
+  
+  aEl.href = res.articles[0].url
+  imgEl.src = res.articles[0].urlToImage
+  authorEl.textContent = res.articles[0].author
+  dateEl.textContent = res.articles[0].publishedAt.substr(0,10);
+  titleEl.textContent = res.articles[0].title
+  contentEl.textContent = res.articles[0].description
+
+  // console.log(cdLink);
+  // console.log(cdInfo);
+
+  for(i=1; i<res.articles.length; i++) {
+    //에러처리
+    let author = res.articles[i].author
+    if (author === null){
+      author = 'author: undefined'
+    }
+    if(/@/g.test(author)){
+      author = 'author: undefined'
+    }
```

Comment:
같은 동작이니 하나의 조건문으로 간소화 가능할 것 같습니다! 또한 `=== null` 구문을 not operator로 한번 더 간소화 가능합니다.
```suggestion
    if (!author || /@/g.test(author)){
      author = 'author: undefined'
    }
```